src/controller/task.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response, NextFunction } from 'express'
+import TaskModel, { Task } from '../model/task'
+
+export default {
+  async createTask (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const { title, description, importance } = req.body
+      const newTask: Task = await TaskModel.create({
+        title,
+        description,
+        importance
+      })
+      res.status(201).json(newTask)
+    } catch (err) {
+      next(err)
+    }
+  },
+  async getAllTasks (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const tasks: Task[] = await TaskModel.find()
+      res.status(200).json({
+        results: tasks.length,
+        tasks
+      })
+    } catch (err) {
+      next(err)
+    }
+  },
+  async getTaskById (req: Request, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const task: Task | null = await TaskModel.findById(req.params.id)
+      res.status(200).json(task)
```

Comment:
Usually I prefer writing like this `res.send(task)` but it's also fine. just fyi